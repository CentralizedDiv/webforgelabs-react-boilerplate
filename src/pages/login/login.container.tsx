import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { LoginRequest } from 'core/services/auth/auth.actions';

import LoginBackground from './login-background/login-background.component';
import LoginForm from './login-form/login-form.component';
import { notification } from 'antd';
import { useTranslation } from 'react-i18next';
import './login.styles.less';
import { AppVersion } from 'core/shared/version/version.component';
import { getAccessToken } from 'routes/auth.router';
import { AuthState } from '../../core/services/auth/auth.models';
import { RouteComponentProps } from 'react-router';

const showLoginError = t => {
  notification.error({
    message: t('alert.login.bad_credentials.title'),
    description: t('alert.login.bad_credentials.description'),
  });
};

export interface ILoginProps extends AuthState, RouteComponentProps {
  // Redux actions
  readonly LoginRequest: any;
}

export const Login: React.FC<ILoginProps> = props => {
  const { t } = useTranslation();

  useEffect(() => {
    if (props.error) showLoginError(t);
  }, [props.error, t]);

  // If successful login
  if (localStorage.getItem('remember') === 'true') {
    if (getAccessToken()) {
      return <Redirect to={'/sample-page'} />;
    }
  } else if (sessionStorage.getItem('access_token')) {
    return <Redirect to={'/sample-page'} />;
  }

  return (
    <div className="st-login-page">
      <LoginForm LoginRequest={props.LoginRequest} isLoading={props.isLoading} error={props.error} />
      <LoginBackground />
      <AppVersion />
    </div>
  );
};

const mapStateToProps = state => state.uiState.auth;

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      LoginRequest,
    },
    dispatch
  );

// Connects the entry-component and makes it the default export.
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
