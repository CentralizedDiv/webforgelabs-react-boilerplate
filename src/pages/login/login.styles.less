.st-login-page {
  display: flex;
  height: 100vh;
  overflow: hidden;

  .st-login-page__login-form-container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex: 0.5;

    .login-form-container__form {
      display: flex;
      flex-direction: column;
      width: 50%;

      .form__logo {
        width: 115px;
        margin-bottom: 40px;
      }

      .form__title {
        margin-bottom: 4px;
        color: @text-color;
        font-size: 20px;
      }

      .form__subtitle {
        color: @text-color-secondary;
        margin-bottom: 32px;
        font-size: 16px;
      }

      .form__inputs {
        .inputs__button {
          width: 100%;
          height: 40px;
        }
        .inputs__forgot {
          float: right;
          color: @text-color-secondary;
          cursor: pointer;

          &:hover {
            color: @primary-color;
          }
        }
        & > .ant-form-item:nth-child(2) {
          margin: 0;
        }
        .inputs__remember {
          margin-bottom: 40px;
        }
      }
    }

    .login-form-container__footer {
      position: absolute;
      width: 100%;
      bottom: 40px;
      display: flex;
      align-items: center;
      justify-content: center;
      color: @text-color-secondary;
      font-style: italic;

      & > span {
        margin-left: 4px;
      }
    }
  }

  .st-login-page__login-background {
    flex: 0.5;
    background-image: url('../../assets/images/bg-login.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    max-width: 50%;
  }
  .welcome-text {
    font-size: 210px;
    font-weight: 500;
    line-height: 137px;
    color: white;
    margin-top: 64vh;
    opacity: 0.2;
    text-transform: uppercase;
    user-select: none;
  }
}
