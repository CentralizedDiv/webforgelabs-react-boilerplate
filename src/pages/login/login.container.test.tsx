import React from 'react';
import { ILoginProps, Login } from './login.container';
import { shallow } from 'enzyme';


describe('Login', () => {
  const mockHistory: any = {
    history: jest.fn(),
    location: jest.fn(),
    match: jest.fn(),
  };

  const props: ILoginProps = {
    ...mockHistory,
    LoginRequest: jest.fn(),
    error: '',
    isLoading: false,
    isLogged: false,
    user: {},
  };
  const component = shallow(<Login {...props} />);

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

});