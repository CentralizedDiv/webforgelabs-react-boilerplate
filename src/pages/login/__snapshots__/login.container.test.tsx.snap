// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `
<div
  className="st-login-page"
>
  <LoginForm
    LoginRequest={[MockFunction]}
    error=""
    isLoading={false}
  />
  <LoginBackground />
  <AppVersion />
</div>
`;
